File: Data Structures & Algorithms/binary-search/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {

        if (nums.length === 1 && nums[0] === target) {
            return 0;
        }

        let minRange = 0;
        let maxRange = nums.length - 1;
        let i = Math.floor((minRange + maxRange) / 2);




        while (minRange < maxRange) {
            console.log(i, nums[i], target);
            if (nums[i] > target) {
                maxRange = i - 1
                i = Math.floor((minRange + maxRange) / 2);
            }

            if (nums[i] < target) {
                minRange = i + 1
                i = Math.floor((minRange + maxRange) / 2);

            }

            if (nums[i] === target) {
                return i
            }
        }

        return -1
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length - 1;

        int[] result = new int[2];

        while(l<r) {
            if(numbers[l] + numbers[r] < target) {
                l = l + 1;
            } else if (numbers[l] + numbers[r] > target) {
                r = r-1;
            } else {
                result[0] = l+1;
                result[1] = r+1;
                break;
            }
        }

        return result;
    }
}
